src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -150,6 +150,14 @@ public static void merge(List<RepoConfiguration> repoConfigs, List<AuthorConfigu
 
             branchMatchingRepoConfig.addAuthors(authorConfig.getAuthorList());
         }
+
+        boolean hasAuthorConfigFile = AuthorConfiguration.DEFAULT_HAS_AUTHOR_CONFIG_FILE;
+        if (!authorConfigs.isEmpty()) {
+            hasAuthorConfigFile = authorConfigs.get(0).hasAuthorConfigFile();
```

Comment:
Can I ask why only the first authorConfig is checked here?

Reply:
Because all the authorConfigs will have the same value of hasAuthorConfigFile. That value is set on line 114 of RepoSense.java. I tried to make hasAuthorConfigFile a static variable of AuthorConfiguration earlier but it caused conflicts between ConfigSystemTests

Reply:
Yes. In line 59, when a RepoConfiguration object is initialized, there is a default AuthorConfiguration attached to it. The line 114 is setting the value for authorConfig found in the CSV, not necessarily matching every authorConfig attached to the repoConfig. Given that repoConfig and authorConfig is matched according to the location and branch, will it possible that the authorConfig of a RepoConfiguration does not have a corresponding CSV record in this case?

Reply:
>  is it possible that the authorConfig of a RepoConfiguration object does not have a corresponding CSV record in this case?

I think that is possible if the user does not include the author for certain locations/branches in the csv. If we move the setAuthorConfigFile() method into the loop, there might be some repoConfigs that has hasAuthorConfigFile = true and the other has hasAuthorConfigFile = false

Reply:
Yes. It seems that your current design is that the inclusion or exclusion of the tagged author depends on whether author-config.csv exists, instead of determining the inclusion on a repository level. In this case, is it possible that you can create a hasAuthorConfig property as a global variable? It may seem a bit strange to attach it to RepoConfiguraiton or AuthorConfiguration

Reply:
I did try to make that property static earlier in commit [75913a3](https://github.com/reposense/RepoSense/pull/1554/checks?sha=75913a3f9aeacd06daa6e5802fbbd101bc36c2b2), but it caused conflicts between config system tests. The tests passed locally but failed on Github CI

Reply:
Can you try again? You can take a look at `LogsManager` for the implementation of the static variable

Reply:
I have changed it to static. Can you take a look?